be/src/udf/udf.cpp

```diff
@@ -145,6 +145,31 @@ bool FunctionContextImpl::check_local_allocations_empty() {
     return false;
 }
 
+void FunctionContextImpl::set_error(const char* error_msg) {
+    std::string* null_error_msg = nullptr;
+    if (_optional_error_msg.compare_exchange_strong(null_error_msg, &_error_msg)) {
+        _error_msg = error_msg;
```

Comment:
If there are two threads th1 and th2.
First&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; -> th1 atomic execute compare_exchange_strong successful, and it stop before "_error_msg = error_msg"
Second -> th2 atomic execute compare_exchange_strong failed(because th1 successful), so it return.
third&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-> th2 execute error_msg to get msg(but it get **undefined content** of _error_msg)
fourth&nbsp;&nbsp;&nbsp;->th1 execute "_error_msg = error_msg" now.

The third step will get undefined content caused by it before fourth, is this may happend?